### ISSUE TITLE
Correção

### ISSUE THREAD

**Issue by luizbossoi** [H4ad/cordova-whatsapp-sticker-plugin#3]
@H4ad nas versões novas do IONIC bem como nas do Android, modificações foram feitas.
Fiz meus testes apenas com Android, porém algumas coisas mudaram.

É necessário adicionar a dependência do androidx e o adapter

ionic cordova plugin add cordova-plugin-androidx
ionic cordova plugin add cordova-plugin-androidx-adapter

Em contents.json, passou a ser obrigatório o parâmetro "image_data_version", por exemplo:
"image_data_version": "1.0",

Com isso o funcionamento ocorrerá normalmente!
Abs

**Comment by H4ad** [comment]
@luizbossoi  Obrigado pela issue.

Eu adicionei as devidas correções, no entanto, sobre as dependências, elas devem ser adicionadas no projeto Ionic é isso ?

**Comment by luizbossoi** [comment]
Isso mesmo, sem as dependências e a correção no JSON, o projeto não roda
Infelizmente não tive como testar no IOS, porém no Android rodou tudo muito bem depois das correções que passei acima na ultima versão do ionic

**Comment by H4ad** [comment]
@luizbossoi Entendi, eu testei os aplicativos que tenho em produção e, tanto o Android quanto o Ios, ainda estão funcionando sem essas modificações, mas já adicionei ao `Readme.md` e `Android.md` as correções no JSON, e assim que conseguir testar as dependências do `androidx` eu adiciono essa correção também.

Obrigado por comentar, irei fechar a issue, se houver mais alguma coisa comente que eu abro ela novamente.

**Comment by fredyepm23** [comment]
I have an assignment from the school I tried but have not succeeded, can I see the ionic source code 3 ,thank you sir.

**Comment by H4ad** [comment]
You tried in what platform? I removed the last source code because contains some sensitive informations, but i wanna create another to leave as example.

When you tried to use, what errors happens?

**Comment by prabhashibuddhima** [comment]
For ionic 4-->

service file

```
import { Injectable } from '@angular/core';
import { IonicNativePlugin, Cordova } from "@ionic-native/core";


// @Plugin({
//   pluginName: 'WhatsAppSticker',
//   plugin: 'cordova-whatsapp-sticker-plugin',
//   pluginRef: 'cordova.plugins.WhatsAppSticker',
//   repo: '',
//   platforms: ['iOS'],
// })

@Injectable({
  providedIn: 'root'
})
export class WhatsappService  extends IonicNativePlugin {

  static pluginName = 'WhatsAppSticker';
  static plugin = 'cordova-whatsapp-sticker-plugin';
  static pluginRef = 'cordova.plugins.WhatsAppSticker';
  static repo = '';
  static platforms = ['iOS'];

  // @Cordova()
  sendToWhatsapp(stickerPack: string): Promise<any> {
    return;
  }
}

```

and you will need to update the webintent by using
`
npm install @ionic-native/web-intent`

And then update app.module.ts and home.page.ts files with (using "/ngx" at the end)

`import { WebIntent } from '@ionic-native/web-intent/ngx';`

**Comment by H4ad** [comment]
@prabhashibuddhima Thanks for the sample, i will try use this sample in ionic 4 and update the README if it works well.

**Comment by prabhashibuddhima** [comment]
@H4ad sorry I have done a mistake. here in the service file ,

```
// @Cordova()
  sendToWhatsapp(stickerPack: string): Promise<any> {
    return;
  }
```

should change to 


```
  sendToWhatsapp(stickerPack: string): Promise<any> {
    return cordova(this,'sendToWhatsapp',{},[]);
  }
```

according to cordova decorator ionic 4. but when I correct it, the plugin gives me an error called,

```

2020-03-03 19:15:13.030672+0530 Nestamolt New Year[827:114166] -[NSNull dataUsingEncoding:]: unrecognized selector sent to instance 0x21c08d9d0
2020-03-03 19:15:13.032984+0530 Nestamolt New Year[827:114166] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[NSNull dataUsingEncoding:]: unrecognized selector sent to instance 0x21c08d9d0'
*** First throw call stack:
(0x1eb810ec4 0x1ea9e1a40 0x1eb729c24 0x1eb8167dc 0x1eb81848c 0x10283f508 0x102842704 0x1028342c8 0x102834130 0x1028364cc 0x1fb554878 0x1fb4f5288 0x1fb4f7ba4 0x1fb21e9f4 0x1fb49ae84 0x1fb1e10e0 0x1fb1e3870 0x1f2b2bd1c 0x1f2b2bfe4 0x1eb7a11f0 0x1eb7a1170 0x1eb7a0a54 0x1eb79b920 0x1eb79b1f0 0x1eda14584 0x218af2d40 0x10282bbe4 0x1eb25abb4)
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb) 
```

Can you help me to sort it out? please..

**Comment by H4ad** [comment]
Maybe you need to send the parameters from function to Cordova.

```Javascript
sendToWhatsapp(stickerPack: string): Promise<any> {
return cordova(this,'sendToWhatsapp', stickerPack,[]);
}
```

I don't know what you need to pass on this function but I think the error it's because you don't pass the data to cordova and the native implementation.

**Comment by prabhashibuddhima** [comment]
@H4d  Hi, I'm still struggling with the same issue

> Maybe you need to send the parameters from function to Cordova.
> 
> ```js
> sendToWhatsapp(stickerPack: string): Promise<any> {
> return cordova(this,'sendToWhatsapp', stickerPack,[]);
> }
> ```
> 
> I don't know what you need to pass on this function but I think the error it's because you don't pass the data to cordova and the native implementation.

Hi, I'm still struggling with that null error. I have tried which you have mentioned here. but still no luck

**Comment by prabhashibuddhima** [comment]
finally, I solved!! after more than one month of struggling.. successfully able to do ith both android and ios. 

@H4d Thanks for this amazing plugin. 

For those are struggling with ios file,

this is my new updated whatsapp.service.ts file

`import { Injectable } from '@angular/core';
import { IonicNativePlugin, cordova } from "@ionic-native/core";


// @Plugin({
//   pluginName: 'WhatsAppSticker',
//   plugin: 'cordova-whatsapp-sticker-plugin',
//   pluginRef: 'cordova.plugins.WhatsAppSticker',
//   repo: '',
//   platforms: ['iOS'],
// })

@Injectable({
  providedIn: 'root'
})
export class WhatsappService  extends IonicNativePlugin {

  static pluginName = 'WhatsAppSticker';
  static plugin = 'cordova-whatsapp-sticker-plugin';
  static pluginRef = 'cordova.plugins.WhatsAppSticker';
  static repo = '';
  static platforms = ['iOS'];

  // @Cordova()
  sendToWhatsapp (stickerPack) :  Promise < any > {
    return  cordova ( this , 'sendToWhatsapp' , {}, [stickerPack]);
  }
}

export interface StickerOptions { 
  ios_app_store_link?: string;


  android_play_store_link?: string;

  
  identifier: string;

  name: string;

  publisher: string;

  
  publisher_website?: string;

 
  privacy_policy_website?: string;

  license_agreement_website?: string;

 
  image_data_version: string;

  
  avoid_cache: boolean;

  
  tray_image: string;

  stickers: StickerImageData[];
}

export interface StickerImageData {  
  
  image_data : string ;    
 
  emojis : string [];    

}`


basically i have changed 

`sendToWhatsapp(stickerPack: string): Promise<any> {
return cordova(this,'sendToWhatsapp', stickerPack,[]);
}`

this to 

```
 sendToWhatsapp (stickerPack) :  Promise < any > {
    return  cordova ( this , 'sendToWhatsapp' , {}, [stickerPack]);
  }
```

cheers!!

**Comment by H4ad** [comment]
I'm glad you managed to solve the problem, good implementation :)

**Comment by stepupithubapp** [comment]
> finally, I solved!! after more than one month of struggling.. successfully able to do in both android and ios.
> 
> @H4ad d Thanks for this amazing plugin.
> 
> For those are struggling with ios file,
> 
> this is my new updated whatsapp.service.ts file
> 
> ```
> import { Injectable } from '@angular/core';
> import { IonicNativePlugin, cordova } from "@ionic-native/core";
> 
> 
> // @Plugin({
> //   pluginName: 'WhatsAppSticker',
> //   plugin: 'cordova-whatsapp-sticker-plugin',
> //   pluginRef: 'cordova.plugins.WhatsAppSticker',
> //   repo: '',
> //   platforms: ['iOS'],
> // })
> 
> @Injectable({
>   providedIn: 'root'
> })
> export class WhatsappService  extends IonicNativePlugin {
> 
>   static pluginName = 'WhatsAppSticker';
>   static plugin = 'cordova-whatsapp-sticker-plugin';
>   static pluginRef = 'cordova.plugins.WhatsAppSticker';
>   static repo = '';
>   static platforms = ['iOS'];
> 
>   // @Cordova()
>   sendToWhatsapp (stickerPack) :  Promise < any > {
>     return  cordova ( this , 'sendToWhatsapp' , {}, [stickerPack]);
>   }
> }
> 
> export interface StickerOptions { 
>   ios_app_store_link?: string;
> 
> 
>   android_play_store_link?: string;
> 
>   
>   identifier: string;
> 
>   name: string;
> 
>   publisher: string;
> 
>   
>   publisher_website?: string;
> 
>  
>   privacy_policy_website?: string;
> 
>   license_agreement_website?: string;
> 
>  
>   image_data_version: string;
> 
>   
>   avoid_cache: boolean;
> 
>   
>   tray_image: string;
> 
>   stickers: StickerImageData[];
> }
> 
> export interface StickerImageData {  
>   
>   image_data : string ;    
>  
>   emojis : string [];    
> 
> }
> ```
> 
> basically i have changed
> 
> ```
>  sendToWhatsapp(stickerPack: string): Promise<any> {
> return cordova(this,'sendToWhatsapp', stickerPack,[]);
>  }
> ```
> 
> this to
> 
> ```
>  sendToWhatsapp (stickerPack) :  Promise < any > {
>     return  cordova ( this , 'sendToWhatsapp' , {}, [stickerPack]);
>   }
> ```
> 
> simply you have to pass stickerPack inside the array.
> 
> @H4ad again thanks a lot for this amazing work! and for the support
> cheers!!

i am also struggling with ionic 3 Platform android

**Comment by H4ad** [comment]
@stepupithubapp Hey man, what's the error that was presented to you? Paste some snippets of code and maybe I can help you.
